tools/test-proxy/Azure.Sdk.Tools.TestProxy.Tests/RecordTests.cs

```diff
@@ -26,7 +27,7 @@ public async Task TestStartRecordSimple()
             httpContext.Request.Body = TestHelpers.GenerateStreamRequestBody(body);
             httpContext.Request.ContentLength = body.Length;
 
-            var controller = new Record(testRecordingHandler)
+            var controller = new Record(testRecordingHandler, new NullLoggerFactory())
```

Comment:
These test changes will be required everywhere. That'll happen after ironing out the core issues with this PR.